### PR TITLE
[IMP] open_academy: Add sessions and instructor field in contact T#59013

### PR DESCRIPTION
**Pull request by geraspace**
- Add new page "sessions" in contact
- Add "instructor" boolean field inside "sessions" page
- Add session records inside "sessions" page

### CODE REVIEW

**Comment by geraspace** [comment]
Hello, @edy1192. This is the inheritance exercise. Thanks!

File: open_academy/models/partner.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from odoo import models, fields
```
Comment (edy1192) [inline open_academy/models/partner.py]: Please rename this file and its references. The standard is 
image and the model is `res.partner` so the file should be `res_partner.py`

File: open_academy/views/partner_views.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (edy1192) [inline open_academy/views/partner_views.xml]: Please rename this file. Should be `res_partner_views.xml`

image

File: open_academy/views/partner_views.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+<odoo>
+
+    <record model="ir.ui.view" id="partner_view_form">
```
Comment (edy1192) [inline open_academy/views/partner_views.xml]: As a good practice we use the same id than original record and add the suffix `_inherit` to indicates that is an inheritance. 

```suggestion
    <record model="ir.ui.view" id="view_partner_form_inherit">
```

File: open_academy/views/partner_views.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        <field name="name">partner.view.form</field>
```
Comment (edy1192) [inline open_academy/views/partner_views.xml]:
```suggestion
        <field name="name">view.partner.form.inherit</field>
```

File: open_academy/models/partner.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    instructor = fields.Boolean(default=False)
```
Comment (edy1192) [inline open_academy/models/partner.py]: Boolean fields are `False` by default, the `default` attribute it's not necessary.

File: open_academy/models/partner.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+from odoo import models, fields
+
+
+class Partner(models.Model):
```
Comment (edy1192) [inline open_academy/models/partner.py]: As a good practice we use the same name that the parent model.

```suggestion
class ResPartner(models.Model):
```

File: open_academy/models/partner.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    _inherit = "res.partner"
+
+    instructor = fields.Boolean(default=False)
+    session_ids = fields.Many2many("open.academy.session", readonly=True)
```
Comment (edy1192) [inline open_academy/models/partner.py]: As a good practice we always place for the name of the relation, the name of both models and the suffix `_rel`. The first name is current model and the second is related model.

```suggestion
    sessions_ids = fields.Many2many("open.academy.session", "res_partner_session_rel")
```

Please apply this comment in `open.academy.session` model to the `many2many` field.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by geraspace** [comment]
I pushed the changes. I believe it's okay now, @edy1192.

**Review by edy1192 — APPROVED** [review]

**Comment by edy1192** [comment]
Hi @geraspace Merged 👍🏼

**Pull request merged**